sympy/physics/units/dimensions.py

```diff
@@ -572,3 +572,51 @@ def is_consistent(self):
         # dimensions
         # in vector language: the set of vectors do not form a basis
         return self.inv_can_transf_matrix.is_square
+    def buckingham_pi_theorem(self, list_of_derived_quantities):
+        """
+        This function is designed to take in a set of physical variables (all essentially dimensionful) and returns one set of all possible exponents,
+        as guided by the Buckingham's pi theorem"""
```

Comment:
Can you add some usage examples? The ones with `>>> ` notation where everyone can see the output. It makes the code easier to read and the documentation gets filled.

Reply:
Thanks for the suggestion. Was a great new experience with docstrings :)